core/src/main/java/org/elasticsearch/ingest/PipelineStore.java

```diff
@@ -104,8 +103,10 @@ void innerUpdatePipelines(ClusterState state) {
         for (PipelineConfiguration pipeline : ingestMetadata.getPipelines().values()) {
             try {
                 pipelines.put(pipeline.getId(), factory.create(pipeline.getId(), pipeline.getConfigAsMap(), processorFactoryRegistry));
+            } catch (ElasticsearchParseException e) {
```

Comment:
problem is that `clusterChanged(ClusterChangeEvent event)` signature does not allow throwing exceptions.

originally it was being caught and threw a `RuntimeException` to get away with it